### PR TITLE
boost: Remove boost dependency

### LINKED ISSUE #174: purge boost dependency
I'd like to open this issue as a reference. In the 1.3.1/1.4 release announcement was a remark to purge boost from VOLK. I consider this issue a place to discuss the required changes.
A quick `grep -irnI boost .` reveals essentially to files which still contain `boost`.

Affected files:
- `lib/qa_utils.h`
`BOOST_AUTO_TEST_CASE` and `BOOST_CHECK_EQUAL`. Since there is no `#include <boost>`, I assume this is just a naming thing. Though, there is also some CMake magic in `lib/CMakeLists.txt`. I suggest to rename these macros and remove the boost reference in the corresponding CMake file.
- `apps/volk_profile.cc`
`boost::filesystem`. This is probably the tough one. C++11 does not support any filesystem magic. All I could find is a pointer to some experimental C++17 features. I guess C++17 is  way to new for VOLK right now. Boost is only used in the function `write_results` with `boost::filesystem::path`, `boost::filesystem::exists` and `boost::filesystem::create_directories`. I suggest to purge the defined namespace `fs` in order to make it more obvious that Boost is used here. And then we need to find an elegant way to remove these function calls. Maybe someone can come up with a good solution here.

### PR DESCRIPTION
**Pull request by jdemel**
VOLK used Boost in the past quite heavily. We reduced Boost usage since then. At the moment, we only require `boost::filesystem` for very old systems. For all recent systems we rely on `std::filesystem` or `std::experimental::filesystem`.

The latest Ubuntu version that still required Boost was 14.04. Starting with Ubuntu 16.04, we can use `std::experimental::filesystem` and later `std::filesystem`.

Fix #174

### CODE REVIEW

**Comment by jdemel** [comment]
So far, all CI tests pass without Boost. However, AppVeyor complains:
```
C:\projects\volk\kernels\volk/volk_32f_sin_32f.h(515): error C3861: '_mm512_floor_ps': identifier not found 
```
This is just an example. It seems like VS 2017 doesn't like to compile AVX512 because it doesn't support it although CMake reports its availability. Thus, we try to move to VS 2019 on AppVeyor as well.

**Comment by marcusmueller** [comment]
> So far, all CI tests pass without Boost. However, AppVeyor complains:
> 
> ```
> C:\projects\volk\kernels\volk/volk_32f_sin_32f.h(515): error C3861: '_mm512_floor_ps': identifier not found 
> ```

> This is just an example. It seems like VS 2017 doesn't like to compile AVX512 because it doesn't support it although CMake reports its availability. Thus, we try to move to VS 2019 on AppVeyor as well.

This seems to be a separate issue and should be tracked separately, I guess?

**Comment by jdemel** [comment]
> > So far, all CI tests pass without Boost. However, AppVeyor complains:
> > ```
> > C:\projects\volk\kernels\volk/volk_32f_sin_32f.h(515): error C3861: '_mm512_floor_ps': identifier not found 
> > ```
> 
> > This is just an example. It seems like VS 2017 doesn't like to compile AVX512 because it doesn't support it although CMake reports its availability. Thus, we try to move to VS 2019 on AppVeyor as well.
> 
> This seems to be a separate issue and should be tracked separately, I guess?

Yes, this is a separate issue.

**Review by michaelld — CHANGES_REQUESTED** [review]
Top-level CMakeLists.txt:29 ... let's just change the CXX_STANDARD to 17 ... no reason to bother with 11 any longer!

File: CMakeLists.txt [hunk 1]
```diff
@@ -195,13 +177,8 @@ if(${FILESYSTEM_FOUND})
     set(CMAKE_CXX_STANDARD 17)
```
Comment (michaelld) [inline CMakeLists.txt]: Per my Request: Let's just move these up to the STANDARD definitions ; that simplifies the current `if/else` to just `if(NOT FILESYSTEM_FOUND` ... no need to `${}` around it either, BTW

Comment (jdemel) [reply CMakeLists.txt]: Agreed. Fixed it.

**Review by michaelld — APPROVED** [review]
LGTM

**Pull request merged**